### ISSUE TITLE
Explore E(2)-equivariant inner model to replace data-augmentation-only symmetry

### ISSUE THREAD

**Issue by nickiliak** [nickiliak/Floorplan-Diffusion-Models#6]
## Description

The current `TransformerModel` denoising network has no built-in equivariance to the E(2) symmetry group (rotations and reflections). Floorplans are inherently symmetric under these transformations, yet the model relies entirely on data augmentation (16x: 4 rotations x 4 flips) to learn approximate symmetry. Replacing the inner model with an E(2)-equivariant architecture would guarantee symmetry at inference, reduce redundant training compute, and improve generation quality.

## Motivation

Floorplans are geometric objects that should be invariant/equivariant under rotations and reflections — a floorplan rotated 90 degrees is equally valid. The current approach handles this by augmenting the training data with all 16 rotation/flip combinations, which:

1. **Wastes training compute** — each symmetry variant is learned independently instead of being a structural guarantee
2. **Only approximates equivariance** — there is no guarantee at inference time; the model can produce symmetry-breaking artifacts
3. **Inflates effective dataset size artificially** — 16x augmentation means 16x more forward/backward passes to cover what an equivariant model gets for free

## Evidence: Current Model is Non-Equivariant

A quantitative equivariance test was run on the current `TransformerModel` (17.8M parameters):

**Test**: Apply a reflection (negate x-coordinates) to input coordinates, then check whether the model output transforms correspondingly.

```
Model: TransformerModel (4-layer, 512-dim, 4-head, triple masked attention)
Total parameters: 17,786,128

Equivariance error (reflection): 0.135854
Output magnitude:                0.116912
Relative equivariance error:     116.20%
```

**Result**: 116% relative equivariance error — the model treats reflected inputs as entirely unrelated. This confirms there is zero structural equivariance; all symmetry knowledge comes from data augmentation.

## Current Architecture

The replacement target is `TransformerModel` at `src/floorplan_diffusion/models/transformer.py:219`:

- **4-layer encoder-only Transformer**, 512-dim hidden, 4 attention heads
- **Triple parallel masked attention** per layer (door / self / general masks) — this is the key architectural innovation from HouseDiffusion
- **Sinusoidal positional encoding** (`transformer.py:34`) — breaks translation equivariance
- **Raw (x,y) coordinate inputs** in [-1, 1], embedded via linear layer (`transformer.py:255`)
- **89-dim condition vector** (room type one-hot + corner/room indices) — inherently invariant, no change needed

### Integration Point

The `create_model()` factory at `scripts/train.py:86-103` is cleanly decoupled from training. An equivariant model can be swapped in by:

1. Adding an `architecture` key to the model config (e.g., `"transformer"` vs `"equivariant"`)
2. Implementing a second factory branch that instantiates the equivariant model
3. The interface contract is: `model(x, timesteps, **cond_kwargs) -> same-shape tensor`

The `FloorplanDiffusionModule` Lightning wrapper (`training/lightning_module.py:21`) is already architecture-agnostic.

### Candidate Architectures

| Architecture | Pros | Cons |
|---|---|---|
| **EGNN** (E(n) Equivariant GNN) | Natural fit for point-based floorplan representation; well-studied | Requires adapting triple-mask attention to message passing |
| **SE(2)-Transformer** | Maintains Transformer structure; closest to current arch | Less mature library support |
| **Vector Neurons** | Can retrofit equivariance onto existing architecture | May not fully preserve triple-mask pattern |
| **escnn / e2cnn** | Well-tested PyTorch E(2) framework; strong library support | General-purpose; may need custom attention layers |

### Key Challenges

1. **Triple-mask attention**: The three parallel attention branches (door/self/general) are the model's core innovation. The equivariant replacement must either incorporate these masks into equivariant message passing or find equivariant alternatives.

2. **Positional encoding replacement**: Sinusoidal positional encoding must be replaced with relative positions or geometric features (pairwise distances, angles).

3. **Coordinate representation**: Raw (x,y) coordinates need to be adapted — E(2)-equivariant models typically operate on relative displacement vectors or invariant scalar features with equivariant vector outputs.

4. **Data augmentation interaction**: If the model is truly equivariant, the 16x augmentation in `dataset.py:170-185` becomes redundant and can be disabled (or kept as regularization).

## Relevant Code

| File | Lines | Component |
|---|---|---|
| `src/floorplan_diffusion/models/transformer.py` | 219-300 | `TransformerModel` — replacement target |
| `src/floorplan_diffusion/models/transformer.py` | 164-216 | `EncoderLayer` — triple masked attention |
| `src/floorplan_diffusion/models/transformer.py` | 34-60 | `PositionalEncoding` — breaks equivariance |
| `scripts/train.py` | 86-103 | `create_model()` — swap point |
| `src/floorplan_diffusion/data/dataset.py` | 170-185 | Data augmentation (rotations + flips) |
| `src/floorplan_diffusion/data/dataset.py` | 371-393 | Mask construction (invariant, no change needed) |
| `src/floorplan_diffusion/training/lightning_module.py` | 21+ | Lightning wrapper (architecture-agnostic) |

## Suggested Tasks

- [ ] Survey equivariant architectures and select the best candidate for 2D floorplan diffusion
- [ ] Implement the equivariant inner model with the same interface contract
- [ ] Adapt or replace the triple-mask attention pattern for equivariance
- [ ] Replace sinusoidal positional encoding with geometric features
- [ ] Add `architecture` config key and factory branch in `create_model()`
- [ ] Benchmark equivariant model vs baseline on training efficiency and sample quality
- [ ] Decide whether to keep or remove data augmentation when using equivariant model

## Additional Context

- No equivariant ML libraries (e2cnn, escnn, e3nn) are currently installed in the project
- The recent notebook commit (`f48fb31`) explores rotational properties of floorplans, suggesting this direction is already being considered
- Evidence files: `artifacts/runs/f3e50d9bd22a893b88069ee57ba53a9a/` contains the equivariance test output and full codebase investigation
